ds3-sdk/src/main/java/com/spectralogic/ds3client/helpers/FileSystemHelperImpl.java

```diff
@@ -0,0 +1,145 @@
+                                                                                    final Path destinationDirectory)
+    {
+        Preconditions.checkNotNull(helpers, "helpers may not be null.");
+        Guard.throwOnNullOrEmptyString(bucketName, "bucketName must have a non-empty value.");
+        Preconditions.checkNotNull(objectNames, "objectNames may not be null.");
+        Preconditions.checkNotNull(destinationDirectory, "destinationDirectory may not be null.");
+
+        long requiredSpace = 0;
+        long availableSpace = 0;
+        final IOException ioException = null;
+
+        if ( ! pathObjectExists(destinationDirectory)) {
+            return new ObjectStorageSpaceVerificationResult(ObjectStorageSpaceVerificationResult.VerificationStatus.PathDoesNotExist,
+                    requiredSpace, availableSpace, ioException);
+        }
+
+        if ( ! pathIsDirectory(destinationDirectory)) {
+            return new ObjectStorageSpaceVerificationResult(ObjectStorageSpaceVerificationResult.VerificationStatus.PathIsNotADirectory,
+                    requiredSpace, availableSpace, ioException);
+        }
+
+        if ( ! pathIsWritable(destinationDirectory)) {
+            return new ObjectStorageSpaceVerificationResult(ObjectStorageSpaceVerificationResult.VerificationStatus.PathLacksAccess,
+                    requiredSpace, availableSpace, ioException);
+        }
+
+        try {
+            helpers.ensureBucketExists(bucketName);
+        } catch (final IOException e) {
+            return new ObjectStorageSpaceVerificationResult(ObjectStorageSpaceVerificationResult.VerificationStatus.BucketDoesNotExist,
+                    requiredSpace, availableSpace, ioException);
+        }
+
+        try {
+            requiredSpace = getRequiredSpaceForObjects(getObjectSizes(helpers, bucketName, objectNames));
+            availableSpace = getAvailableFileSpace(destinationDirectory);
+        } catch (final IOException e) {
+            return new ObjectStorageSpaceVerificationResult(ObjectStorageSpaceVerificationResult.VerificationStatus.CaughtIOException,
+                    requiredSpace, availableSpace, e);
+        }
+
+        final ObjectStorageSpaceVerificationResult.VerificationStatus verificationStatus = availableSpace > requiredSpace ?
+                ObjectStorageSpaceVerificationResult.VerificationStatus.OK : ObjectStorageSpaceVerificationResult.VerificationStatus.PathLacksSufficientStorageSpace;
+
+        return new ObjectStorageSpaceVerificationResult(verificationStatus, requiredSpace, availableSpace, ioException);
+    }
+
+    private Map<String, Long> getObjectSizes(final Ds3ClientHelpers helpers,
+                                             final String bucketName,
+                                             final Collection<String> objectNames)
+                                             throws IOException
+    {
+        final Map<String, Long> objectSizeMap = new HashMap<>();
```

Comment:
Is this map going to be mutated?  If not convert this to an ImmutableMap and use the ImmutableMap.Builder to create it.